### PR TITLE
command: print track metadata in ${track-list}

### PR DESCRIPTION
**Pull request by guidocella**
Extract track metadata formatting out of loadfile.c to reuse in the track-list property.

### CODE REVIEW

File: player/command.c [hunk 1]
```diff
@@ -2104,21 +2098,70 @@ static const char *track_type_name(struct track *t)
...
+    if (t->visual_impaired_track)
+        ADD_FLAG(ctx, dst, "visual-impaired", first);
+    if (t->hearing_impaired_track)
+        ADD_FLAG(ctx, dst, "hearing-impaired", first);
+    if (t->is_external)
+        ADD_FLAG(ctx, dst, "external", first);
+    if (!first)
+        bstr_xappend0(ctx, &dst, "]");
+
+    return bstrto0(ctx, dst);
```
Comment (kasper93) [inline player/command.c:2162]: This could return bstr.

File: player/command.h [hunk 2]
```diff
@@ -121,4 +122,10 @@ void mark_seek(struct MPContext *mpctx);
 
 void mp_abort_cache_dumping(struct MPContext *mpctx);
 
+// U+25CB WHITE CIRCLE
+// U+25CF BLACK CIRCLE
+#define WHITE_CIRCLE "\xe2\x97\x8b"
+#define BLACK_CIRCLE "\xe2\x97\x8f"
```
[the remark this answers is not in the document]
Comment (guidocella) [reply player/command.h:128]: It's moved to the header file to reuse it in loadlist.c.

File: player/command.c (hunk 1 above)
Comment (guidocella) [reply player/command.c:2162]: Neither of the callers use bstr though.

File: player/command.c [hunk 3]
```diff
@@ -2104,21 +2098,70 @@ static const char *track_type_name(struct track *t)
...
     return res;
 }
 
+#define bstr_xappend0(ctx, dst, s) bstr_xappend(ctx, dst, bstr0(s))
+#define ADD_FLAG(ctx, dst, flag, first) do {                           \
+    bstr_xappend_asprintf(ctx, &dst, " %s%s", first ? "[" : "", flag); \
+    first = false;                                                     \
+} while(0)
+
+char *mp_format_track_metadata(void *ctx, struct track *t)
```
[the remark this answers is not in the document]
Comment (guidocella) [reply player/command.c:2116]: Yeah but the 2 callers append differently because `print_stream` uses a fixed length buffer.

**Review by kasper93 — APPROVED** [review]
Works.

**Pull request merged**